tools/doc/html.js

```diff
@@ -93,12 +93,12 @@ function toHTML({ input, content, filename, nodeVersion, versions }) {
 // Set the section name based on the first header.  Default to 'Index'.
 function firstHeader() {
   return (tree, file) => {
-    file.section = 'Index';
-
     const heading = find(tree, { type: 'heading' });
-    if (heading) {
-      const text = find(heading, { type: 'text' });
-      if (text) file.section = text.value;
+
+    if (heading && heading.children.length) {
+      file.section = heading.children.map(({ value }) => value).join('');
+    } else {
+      file.section = 'Index';
```

Comment:
Thanks for making this PR &mdash; I hadn't noticed this bug yet, but this patch does indeed fix it.

It's not immediately obvious what is going on here. Would you consider adding a comment along w/ this patch?

For someone to understand this, the developer would need to be familiar w/ the Remark AST and even then, it's not obvious why wrapping each value and joining their string representation results in the intended document title.

Reply:
I've updated the OP with an explanation of the new code, PTAL